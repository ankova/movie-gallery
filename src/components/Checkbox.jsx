import React, { Fragment } from 'react';

const Checkbox = (props) => {
    const {name, onChange} = props;

    return (
        <Fragment>
            
            <label>
                <input type="checkbox" value={name} onChange={onChange}/>
                {name} 
            </label>
        </Fragment>
    )
}

export default Checkbox;