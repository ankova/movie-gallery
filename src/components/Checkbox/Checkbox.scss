/*TO DO
//Add styling for checkbox
*/