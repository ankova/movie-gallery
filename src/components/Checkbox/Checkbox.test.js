import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
    let component;
    const onChange = jest.fn();
    const props = {
        name: 'action',
        onChange
    };

    it('renders component without crashing', () => {
        const tree = renderer.create(<Checkbox />).toJSON;
        expect(tree).toMatchSnapshot();
    });

    beforeEach(() => component = mount(<Checkbox {...props} />));

    it('checks if onChange works', () => {
        component.find('input').simulate('change');
        expect(onChange).toBeCalled();
    });
});
