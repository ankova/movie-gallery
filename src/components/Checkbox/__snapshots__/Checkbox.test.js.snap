exports[`Checkbox renders component without crashing 1`] = `[Function]`;
